clock.go

```diff
@@ -28,6 +31,7 @@ func (c *SystemClock) Nanos() int64 {
 // source. You need to seed the time with either .SetFromDuration() or
 // .SetNanos().
 type ManualClock struct {
+	mu    sync.RWMutex
```

Comment:
We should just use `sync/atomic` and do `LoadInt64(&nanos)/StoreInt64(&nanos, value)` to make it thread-safe. We don't need for a read-write mutex.

Reply:
Works for me. I've amended that commit and removed the updates to the tests since field order is no longer important.